vcs-diff-lint-csdiff-ruff

```diff
@@ -24,16 +24,32 @@ def ruff_check():
     return json.loads(out)
 
 
-def ruff_code_to_name(code):
+def load_human_readable_map():
     """
-    Convert noqa code e.g. F401 to its human-readable name, e.g. unused-import
+    Introspect ruff and map all possible noqa codes to a human-readable names.
     """
-    # This implementation will likely kill all ruff performance benefits
-    cmd = ["ruff", "rule", code, "--output-format=json"]
+    the_map = getattr(load_human_readable_map, "map", None)
+    if the_map is not None:
+        return the_map
```

Comment:
I really like the idea of fetching all noqa codes at once, the performance improvement will be huge.

However, instead of this DYI caching, I'd rather see the `@lru_cache` decorator or maybe better yet, dropping the `ruff_code_to_name` function and calling this `load_human_readable_map` only once in main.

Reply:
Good idea, updated!